lib/nokogiri/xml/node_set.rb

```diff
@@ -293,7 +293,11 @@ def == other
       # Returns a new NodeSet containing all the children of all the nodes in
       # the NodeSet
       def children
-        inject(NodeSet.new(document)) { |set, node| set += node.children }
```

Comment:
shouldn't be necessary.
wanted to make sure only a single `NodeSet` is created (as I was debugging how the whole piece behaves with decorators), than I left it in as it seemed to make sense but I do not have benchmarks.

it "must" be faster :) but you're right this should have been 2 PRs I just :nail_care:  as I went with the fixes.

on my defense - its uglier but similar to the `def reverse` code bellow :wink: 
I am also fine with moving it out ... if you prefer to have it separately?